### ISSUE TITLE
Accessing the Jupyter Server API when using Jupyterhub

### ISSUE THREAD

**Issue by philejmath** [jupyterhub/jupyterhub#1078]
The documentation is unclear on how to access the [Jupyter Notebook API
](https://github.com/jupyter/jupyter/wiki/Jupyter-Notebook-Server-API) when using Jupyterhub. 

[Jupyterhub Documentation](http://jupyterhub.readthedocs.io/en/latest/rest.html) states : 
```
Note that the token authorizes JupyterHub REST API requests.
 The same token does not authorize access to the Jupyter Notebook REST API provided by notebook servers managed by JupyterHub.
```
I setup a system wide jupyter_notebook_config.py in /etc/jupyter using "c.NotebookApp.token" expecting to be able to access the notebook API as specified in the docs. 
```
curl -H "Authorization: Token TOKEN" -H "Content-Type: application/json"  -XGET http://localhost:8000/user/username/api
```
All attempts to do so seem to push me to the Jupyterhub login page which is confusing since the jupyterhub documentation states that you cannot authenticate to the Notebook REST API with a Jupyterhub token.

**Comment by minrk** [comment]
In JupyterHub 0.7, there is no mechanism for token authentication to single-user servers. Only cookies can be used. 0.8 supports using JupyterHub API tokens (#961), so we should make sure to update the docs to reflect that.

**Comment by mrunmayeejog** [comment]
Currently using jupyterhub 0.7.0 I am facing the similar issue. I can't upgrade to 0.8.0 version as of now. 
I created a new API token usin jupyterhub token username and added it jupyterhub configs file. With this I am able to access jupyterhub REST APIs. Same token can not be used for accessing jupyter notebook apis as mentioned above.

Regarding authentication using cookies I have tried the following. I am testing APIs using Python's requests package. eg: 
url = "http://localhost:8000/user/<username>/api/contents"
req = requests.get(url=url, cookies=<cookie_secret>)

This cookie_secret I am currently reading from the secret_file. Now I am not sure and how to pass the cookie for these APIs.
